src/types/UiaTextRangeBase.cpp

```diff
@@ -501,42 +501,42 @@ IFACEMETHODIMP UiaTextRangeBase::GetText(_In_ int maxLength, _Out_ BSTR* pRetVal
     // truncated.
     const bool getPartialText = maxLength != -1;
 
-    if (!_degenerate)
+    if (!IsDegenerate())
     {
         try
         {
-            const ScreenInfoRow startScreenInfoRow = _endpointToScreenInfoRow(_pData, _start);
-            const Column startColumn = _endpointToColumn(_pData, _start);
-            const ScreenInfoRow endScreenInfoRow = _endpointToScreenInfoRow(_pData, _end);
-            const Column endColumn = _endpointToColumn(_pData, _end);
-            const unsigned int totalRowsInRange = _rowCountInRange(_pData);
-            const TextBuffer& textBuffer = _pData->GetTextBuffer();
-
 #if defined(_DEBUG) && defined(UIATEXTRANGE_DEBUG_MSGS)
             std::wstringstream ss;
-            ss << L"---Initial span start=" << _start << L" and end=" << _end << L"\n";
-            ss << L"----Retrieving sr:" << startScreenInfoRow << L" sc:" << startColumn << L" er:" << endScreenInfoRow << L" ec:" << endColumn << L"\n";
+            ss << L"---Initial span start={" << _start.X << L", " << _start.Y << L"} and end={" << _end.X << ", " << _end.Y << L"}\n";
             OutputDebugString(ss.str().c_str());
 #endif
 
+            // if _end is at 0, we ignore that row because _end is exclusive
+            const auto& buffer = _pData->GetTextBuffer();
+            const unsigned int totalRowsInRange = (_end.X == buffer.GetSize().Left()) ?

```

Comment:
Why is this and the loop variable below an `unsigned int`? Isn't all of this math being done with `SHORT`s that are a part of `COORD`s?

Reply:
just a remnant of what used to be. Fixed!